src/tests/Model/ResultSummaryReporterTests.cs

```diff
@@ -0,0 +1,135 @@
+            Assert.That(report, Does.Contain("Warnings: 3"));
+            Assert.That(report, Does.Contain("Inconclusive: 4"));
+            Assert.That(report, Does.Contain("Skipped: 5"));
+        }
+
+        [Test]
+        public void ReportDoesNotContainExtendedFailureInformationWhenNoTestsFailed()
+        {
+            var summary = new ResultSummary
+            {
+                FailureCount = 0,
+                ErrorCount = 0,
+                InvalidCount = 0,
+                OverallResult = String.Empty
+            };
+
+            var report = ResultSummaryReporter.WriteSummaryReport(summary);
+            Assert.That(report, Does.Not.Contain("Failed Tests - Failures:"));
+            Assert.That(report, Does.Not.Contain("Errors:"));
+            Assert.That(report, Does.Not.Contain("Invalid:"));
+        }
+
+        [TestCase(1,0,0)]
+        [TestCase(0,1,0)]
+        [TestCase(0,0,1)]
+        public void ReportContainsExtendedFailureInformationWhenItHasFailures(int failureCount, int errorCount, int invalidCount)
+        {
+            var summary = new ResultSummary
+            {
+                FailureCount = failureCount,
+                ErrorCount = errorCount,
+                InvalidCount = invalidCount,
+                OverallResult = String.Empty
+            };
+
+            var report = ResultSummaryReporter.WriteSummaryReport(summary);
+            Assert.That(report, Does.Contain("Failed Tests - Failures:"));
+            Assert.That(report, Does.Contain("Errors:"));
+            Assert.That(report, Does.Contain("Invalid:"));
+        }
+
+        [Test]
+        public void ReportDoesNotContainExtendedSkipInformationWhenNoTestsAreSkipped()
+        {
+            var summary = new ResultSummary
+            {
+                IgnoreCount = 0,
+                ExplicitCount = 0,
+                SkipCount = 0,
+                OverallResult = String.Empty
+            };
+
+            var report = ResultSummaryReporter.WriteSummaryReport(summary);
+
+            Assert.That(report, Does.Not.Contain("Skipped Tests - Ignored:"));
+            Assert.That(report, Does.Not.Contain("Explicit:"));
+            Assert.That(report, Does.Not.Contain("Other:"));
+        }
+
+        [TestCase(1,0,0)]
+        [TestCase(0,1,0)]
+        [TestCase(0,0,1)]
+        public void ReportContainsExtendedSkipInformationWhenItHasSkips(int ignoreCount, int explicitCount, int skipCount)
+        {
+            var summary = new ResultSummary
+            {
+                IgnoreCount = ignoreCount,
+                ExplicitCount = explicitCount,
+                SkipCount = skipCount,
+                OverallResult = String.Empty
+            };
+
+            var report = ResultSummaryReporter.WriteSummaryReport(summary);
+
+            Assert.That(report, Does.Contain("Skipped Tests - Ignored:"));
+            Assert.That(report, Does.Contain("Explicit:"));
+            Assert.That(report, Does.Contain("Other:"));
```

Comment:
I think that we should assert that the correct counts are presented in the report.